### DISCUSSION TITLE
Help with the typescript rewriting

### DISCUSSION THREAD

**Discussion by DevWannabe-dot** [qsniyg/maxurl#890] [Q&A] [answered]
I've never tried a discussion in github before soo if there is anything wrong I can create an issue instead! I know there is a lot to fix and didn't mean to overload the issues queue, but, I've been having some problems with the [userscript.ts](https://github.com/qsniyg/maxurl/blob/master/src/userscript.ts). I tried sideloading the firefox extension and also the [userscript_smaller.user.js](https://github.com/qsniyg/maxurl/blob/master/userscript_smaller.user.js) that is used on the github.io page, but they seem not to be updated, so I've got to go with either the typescript or the javascript userscript that are updated frequently, right?

My questions are, will the javascript version lose support soon? And how do I get the .ts to work? I tried both violentmonkey and tampermonkey but it's not doing anything in any? I tried copying and pasting and installing from url. It does match all the sites but doesn't work, I checked in the github page and it could give me the max url, while the updated script did nothing? Just need some advice, thanks!

**Comment by qsniyg** [comment, marked as answer]
Sorry, yeah I haven't yet updated the documentation properly yet, I still haven't had a chance to finish updating the build tools (hence why I haven't yet merged #861).

The typescript rewrite only affects the source code. The end result (once compiled through `tsc`) is still `userscript.user.js`, which is the one you have to load through VM/TM or the extension.

`userscript_smaller.user.js` is traditionally only updated on new releases, so it's used as a sort of permalink for the stable version. In retrospect I should've used a tag or branch or something instead...

In order to generate `userscript.user.js`, simply run `tsc` (if the typescript compiler is installed), and it should compile it for you. If you want to generate `userscript_smaller.user.js` (or any of the files under build), run `node tools/remcomments.js nowatch`.

I hope this helps! :)

(I'm also new to this discussion thing too haha)

**Comment by DevWannabe-dot** [comment]
Big ups, dude! Now I understand it, thank you!
